### DISCUSSION TITLE
dev environment for serveless APIs

### DISCUSSION THREAD

**Discussion by aegatlin** [vercel/vercel#5889] [Help]
Hello,

Short version: What is the best way to setup a dev environment for a Vercel serverless JSON API that does _not_ have a frontend?

Long version: I am exploring deploying a serverless JSON API that is a standalone microservice with Vercel (using Node.js + Typescript). I am not using Nextjs or another frontend library/framework within this project, although it will be consumed by a few other projects in the long run. I have not been able to find information on the best way to get a dev environment going on my localhost when the project is _only_ an Vercel API, i.e., it's _only_ an `/api` folder and _not_ also a Nextjs app, etc. (so, no frontend at all). I tried `vercel dev` but soon realized that was designed for some other purpose. I'm not sure which docs page to read for this, or what the recommended approach is. Is there a `{vercel,now,next,?} dev` CLI that has it "just works"?

**Comment by mikestopcontinues** [comment]
Did you hit on a solution to this? I know I can just fall back on one of the minimal templates, but I'd really prefer not to.

**Reply by aegatlin** [reply]
I wound up going with a vercel app that does have an index.html that just has nothing on it. Then just using the api layer. My question is essentially asking for a "clean solution" to API-only serverless by vercel, and I think the answer is "there is none". But, I hope that doesn't sound (too) critical, because the solution of having a noop index.html is still pretty darn clean. This is a toy-app, but it is a fully functioning dictionary live and serving dictionary definitions on my personal website: https://github.com/aegatlin/marmee

**Reply by mikestopcontinues** [reply]
That's a neat project! I have it set up with Next now, though that method bundles all the endpoints into one. I was looking for a better solution since the underlying serverless func logic doesn't merge, and I've got some pretty heavy endpoints (scraping, image processing), I was hoping to split off from the others. I guess I can go with microservices. :)

**Reply by aegatlin** [reply]
My main desire for attempting this API-only microservice in Vercel was that I _didn’t_ want to deal with all the hassle of AWS Lambda or GCP Compute Engine (or whatever it’s called). I wanted the “Vercel experience”, but just for backend-only microservices. It wasn’t as easy as I hoped (I think Vercel as a company just isn’t prioritizing this because they are more focused on static web and not trying to be a Platform as a Service, etc.), but, Vercel dev experience is soo much better than the major platforms that I’d rather just use Vercel in this weird way, lol.

Good luck! :D
